jdaviz/core/template_mixin.py

```diff
@@ -154,38 +155,59 @@ def show_widget(widget, loc, title):  # pragma: no cover
         raise ValueError(f"Unrecognized display location: {loc}")
 
 
+def _is_spectrum_viewer(viewer):
+    return ('ProfileView' in viewer.__class__.__name__
+            or viewer.__class__.__name__ == 'Spectrum1DViewer')
+
+
+def _is_spectrum_2d_viewer(viewer):
+    return ('Profile2DView' in viewer.__class__.__name__
+            or viewer.__class__.__name__ == 'Spectrum2DViewer')
+
+
+def _is_image_viewer(viewer):
+    return 'ImageView' in viewer.__class__.__name__
+
+
 class ViewerPropertiesMixin:
     # assumes that self.app is defined by the class
-    @cached_property
+    @property
     def spectrum_viewer(self):
-        if hasattr(self, '_default_spectrum_viewer_reference_name'):
-            viewer_reference = self._default_spectrum_viewer_reference_name
-        else:
-            viewer_reference = self.app._get_first_viewer_reference_name(
-                require_spectrum_viewer=True
-            )
+        viewer_reference = self.app._get_first_viewer_reference_name(
+            require_spectrum_viewer=True
+        )
+        if viewer_reference is None:
+            return None
 
         return self.app.get_viewer(viewer_reference)
 
-    @cached_property
+    @property
```

Comment:
we could/should re-implement these list properties as cached properties.  See https://github.com/spacetelescope/jdaviz/pull/3473#discussion_r2022924783